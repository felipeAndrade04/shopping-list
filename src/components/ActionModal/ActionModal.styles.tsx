import styled from 'styled-components/native';

export const Container = styled.View`
  width: 100%;
  padding: 24px;
  align-items: center;
`;

export const Title = styled.Text`
  font-size: 18px;
  font-family: ${({ theme }) => theme.fonts.medium};
`;

export const Actions = styled.View`
  flex-direction: row;
`;

export const Action = styled.View`
  flex: 1;
`;
