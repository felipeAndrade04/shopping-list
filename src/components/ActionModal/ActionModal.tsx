import React, { useState } from 'react';
import { Modal } from '../Modal';
import { ActionModalProps } from './ActionModal.types';
import * as S from './ActionModal.styles';
import { Input } from '../Input';
import { Button } from '../Button';
import { colors } from '@app/theme';
import { Spacer } from '../Spacer';

export function ActionModal({
  show,
  title,
  successActionText,
  isLoading,
  close,
  successAction,
}: ActionModalProps) {
  const [name, setName] = useState('');

  async function handleSuccess() {
    await successAction(name);
    setName('');
  }

  return (
    <Modal show={show} close={close}>
      <S.Container>
        <S.Title>{title}</S.Title>
        <Spacer dimesion={24} />
        <Input
          value={name}
          onChangeText={(event) => setName(event)}
          placeholder="Informe o nome"
          borderColor={colors.dark}
          borderWidth={1}
        />
        <Spacer dimesion={48} />
        <S.Actions>
          <S.Action>
            <Button onPress={close} variant="outline" color={colors.red} testID="cancelButton">
              Cancelar
            </Button>
          </S.Action>
          <Spacer dimesion={16} />
          <S.Action>
            <Button onPress={handleSuccess} isLoading={isLoading} testID="successButton">
              {successActionText}
            </Button>
          </S.Action>
        </S.Actions>
      </S.Container>
    </Modal>
  );
}
