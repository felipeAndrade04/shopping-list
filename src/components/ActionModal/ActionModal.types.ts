export interface ActionModalProps {
  show: boolean;
  title: string;
  successActionText: string;
  successAction: () => void;
  close: () => void;
}
