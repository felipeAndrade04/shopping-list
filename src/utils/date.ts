const months = ['jan', 'fev', 'mar', 'abr', 'mai', 'jun', 'jul', 'ago', 'set', 'out', 'nov', 'dez'];

export function formatDate(date: Date) {
  const day = date.getDay();
  const month = date.getMonth();

  return `${day} de ${months[month]}`;
}
