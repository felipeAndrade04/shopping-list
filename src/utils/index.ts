export * from './delay';
export * from './date';
