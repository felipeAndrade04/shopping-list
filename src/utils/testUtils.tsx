import React from 'react';
import { authReducer } from '@app/store';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components/native';
import * as theme from '@app/theme';
import { configureStore } from '@reduxjs/toolkit';

let store = configureStore({
  reducer: {
    auth: authReducer,
  },
});

export const wrapper = ({ children }: { children: ReactNode }) => (
  <ThemeProvider theme={theme}>
    <Provider store={store}>{children}</Provider>
  </ThemeProvider>
);

export function resetStore() {
  store = configureStore({
    reducer: {
      auth: authReducer,
    },
  });
}
