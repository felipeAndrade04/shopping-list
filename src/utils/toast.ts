import EventManager from './EvenManager';

export type ToastType = 'success' | 'info' | 'error';

export interface Toast {
  id?: string;
  type: ToastType;
  text: string;
  duration?: number;
}

export const toastEventManager = new EventManager();

export default function toast({ type, text, duration }: Toast) {
  toastEventManager.emit<Toast>('addtoast', {
    id: new Date().getTime().toString(),
    type,
    text,
    duration,
  });
}
