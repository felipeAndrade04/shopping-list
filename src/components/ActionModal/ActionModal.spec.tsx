import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import { wrapper } from '@app/utils/testUtils';
import { ActionModal } from './ActionModal';

describe('<ActionModal />', () => {
  const handleSuccess = jest.fn(() => Promise.resolve());
  const title = 'Modal Text';

  it('Should render correctly', () => {
    const tree = render(
      <ActionModal
        show={true}
        close={jest.fn}
        successAction={handleSuccess}
        title="Modal Test"
        successActionText="Test"
      />,
      {
        wrapper,
      }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('Should show content on screen', () => {
    const { queryByText } = render(
      <ActionModal
        show={true}
        close={jest.fn}
        successAction={handleSuccess}
        title={title}
        successActionText="Test"
      />,
      { wrapper }
    );

    expect(queryByText(title)).toBeTruthy();
  });

  it('Shold close modal on overlay click', () => {
    let show = true;

    const { queryByTestId } = render(
      <ActionModal
        show={show}
        close={() => (show = false)}
        successAction={handleSuccess}
        title={title}
        successActionText="Test"
      />,
      { wrapper }
    );

    expect(show).toBeTruthy();

    const overlay = queryByTestId('overlay');
    fireEvent.press(overlay);

    expect(show).toBeFalsy();
  });

  it('Shold close modal on cancel button click', () => {
    let show = true;

    const { queryByTestId } = render(
      <ActionModal
        show={show}
        close={() => (show = false)}
        successAction={handleSuccess}
        title={title}
        successActionText="Test"
      />,
      { wrapper }
    );

    expect(show).toBeTruthy();

    const cancelButton = queryByTestId('cancelButton');
    fireEvent.press(cancelButton);

    expect(show).toBeFalsy();
  });

  it('Shold call close function on overlay click', () => {
    const close = jest.fn();

    const { queryByTestId } = render(
      <ActionModal
        show={true}
        close={close}
        successAction={handleSuccess}
        title={title}
        successActionText="Test"
      />,
      { wrapper }
    );

    const overlay = queryByTestId('overlay');
    fireEvent.press(overlay);

    expect(close).toBeCalledTimes(1);
  });

  it('Shold call close function on cancel button click', () => {
    const close = jest.fn();

    const { queryByTestId } = render(
      <ActionModal
        show={true}
        close={close}
        successAction={handleSuccess}
        title={title}
        successActionText="Test"
      />,
      { wrapper }
    );

    const cancelButton = queryByTestId('cancelButton');
    fireEvent.press(cancelButton);

    expect(close).toBeCalledTimes(1);
  });

  it('Shold call success function', () => {
    const { queryByTestId } = render(
      <ActionModal
        show={true}
        close={jest.fn}
        successAction={handleSuccess}
        title={title}
        successActionText="Test"
      />,
      { wrapper }
    );

    const successButton = queryByTestId('successButton');
    fireEvent.press(successButton);

    expect(handleSuccess).toBeCalledTimes(1);
  });
});
