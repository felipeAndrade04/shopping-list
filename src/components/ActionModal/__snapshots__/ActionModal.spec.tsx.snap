// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionModal /> Should render correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  style={
    Array [
      Object {},
    ]
  }
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#363F5Faa",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
        "width": "100%",
      }
    }
    testID="overlay"
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#F4F4F4",
          "borderRadius": 16,
          "opacity": 1,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "paddingBottom": 24,
              "paddingLeft": 24,
              "paddingRight": 24,
              "paddingTop": 24,
              "width": "100%",
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins-Medium",
                "fontSize": 18,
              },
            ]
          }
        >
          Modal Test
        </Text>
        <View
          dimesion={24}
          style={
            Array [
              Object {
                "height": 24,
                "width": 24,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "width": "100%",
              },
            ]
          }
        >
          <View
            borderColor="#0D0A0B"
            borderWidth={1}
            isFocused={false}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#0D0A0B",
                  "borderRadius": 6,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "opacity": 1,
                  "paddingBottom": 0,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 0,
                  "transition": "border-color 0.3s ease-in",
                  "width": "100%",
                },
              ]
            }
          >
            <TextInput
              editable={true}
              onBlur={[Function]}
              onChangeText={[Function]}
              onFocus={[Function]}
              placeholder="Informe o nome"
              style={
                Array [
                  Object {
                    "color": "#0D0A0B",
                    "flexBasis": 0,
                    "flexGrow": 1,
                    "flexShrink": 1,
                    "fontFamily": "Poppins-Regular",
                    "fontSize": 16,
                    "height": 50,
                    "paddingBottom": 8,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "paddingTop": 8,
                  },
                ]
              }
              testID="input"
              value=""
            />
          </View>
        </View>
        <View
          dimesion={48}
          style={
            Array [
              Object {
                "height": 48,
                "width": 48,
              },
            ]
          }
        />
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <RNGestureHandlerButton
              activeOpacity={0.105}
              collapsable={false}
              color="#FE5F55"
              delayLongPress={600}
              enabled={true}
              handlerTag={1}
              handlerType="NativeViewGestureHandler"
              onActiveStateChange={[Function]}
              onGestureEvent={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onHandlerStateChange={[Function]}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#FE5F55",
                  "borderRadius": 6,
                  "borderStyle": "solid",
                  "borderWidth": 2,
                  "display": "flex",
                  "height": 56,
                  "justifyContent": "center",
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "opacity": 1,
                  "paddingBottom": 8,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 8,
                  "width": "100%",
                }
              }
              testID="cancelButton"
              underlayColor="black"
              variant="outline"
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "black",
                    "borderBottomLeftRadius": undefined,
                    "borderBottomRightRadius": undefined,
                    "borderRadius": 6,
                    "borderTopLeftRadius": undefined,
                    "borderTopRightRadius": undefined,
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
              <Text
                color="#FE5F55"
                style={
                  Array [
                    Object {
                      "color": "#FE5F55",
                      "fontFamily": "Poppins-Bold",
                      "fontSize": 18,
                    },
                  ]
                }
                variant="outline"
              >
                Cancelar
              </Text>
            </RNGestureHandlerButton>
          </View>
          <View
            dimesion={16}
            style={
              Array [
                Object {
                  "height": 16,
                  "width": 16,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                },
              ]
            }
          >
            <RNGestureHandlerButton
              activeOpacity={0.105}
              collapsable={false}
              color="#009Cf7"
              delayLongPress={600}
              enabled={true}
              handlerTag={2}
              handlerType="NativeViewGestureHandler"
              onActiveStateChange={[Function]}
              onGestureEvent={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onHandlerStateChange={[Function]}
              onPress={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#009Cf7",
                  "borderColor": "black",
                  "borderRadius": 6,
                  "borderStyle": "solid",
                  "borderWidth": 0,
                  "display": "flex",
                  "height": 56,
                  "justifyContent": "center",
                  "marginBottom": 0,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 0,
                  "opacity": 1,
                  "paddingBottom": 8,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 8,
                  "width": "100%",
                }
              }
              testID="successButton"
              underlayColor="black"
              variant="solid"
            >
              <View
                collapsable={false}
                style={
                  Object {
                    "backgroundColor": "black",
                    "borderBottomLeftRadius": undefined,
                    "borderBottomRightRadius": undefined,
                    "borderRadius": 6,
                    "borderTopLeftRadius": undefined,
                    "borderTopRightRadius": undefined,
                    "bottom": 0,
                    "left": 0,
                    "opacity": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              />
              <Text
                color="#009Cf7"
                style={
                  Array [
                    Object {
                      "color": "#FFFFFF",
                      "fontFamily": "Poppins-Bold",
                      "fontSize": 18,
                    },
                  ]
                }
                variant="solid"
              >
                Test
              </Text>
            </RNGestureHandlerButton>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
